src/fidesops/service/privacy_request/request_runner_service.py

```diff
@@ -281,6 +282,15 @@ def run_privacy_request(
             return
 
         privacy_request.finished_processing_at = datetime.utcnow()
+        AuditLog.create(
+            db=session,
+            data={
+                "user_id": "system",
+                "privacy_request_id": privacy_request.id,
+                "action": AuditLogAction.finished,
+                "message": "",
+            },
+        )
```

Comment:
Something that would be nice here is to wrap this new `AuditLog` creation into a transaction with `privacy_request.save(...)` — if one of these operations fails without the other we'd end up with an incomplete state, so it's better to commit them both at once.

Reply:
Good find. That is a potential issue when approved/denied are created as well. All of the audit logs should probably be updated at the same time when this is done.

Would you like this to be addressed in this PR or a follow up?

Reply:
Let's do it as a [follow-up](https://github.com/ethyca/fidesops/issues/1057)